src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReplacementService.cs

```diff
@@ -111,20 +127,30 @@ public class ReplacementService : IReplacementService
 
                     // This Release Id can be found on the ReplacementFileReference
                     var releaseId = (await _contentDbContext.ReleaseFileReferences
-                        .FindAsync(replacementReleaseFileReferenceId)).ReleaseId;
+                        .FindAsync(replacementFileReferenceId)).ReleaseId;
 
                     return await RemoveSubjectAndFileFromRelease(releaseId, replacementPlan.OriginalSubjectId,
-                        originalReleaseFileReferenceId);
+                        originalFileReferenceId);
                 });
         }
 
         private async Task<Either<ActionResult, ReleaseFileReference>> CheckReleaseFileReferenceExists(Guid id)
         {
-            return await _contentPersistenceHelper.CheckEntityExists<ReleaseFileReference>(id)
-                .OnSuccess(releaseFileReference => releaseFileReference.ReleaseFileType != ReleaseFileTypes.Data
-                    ? new Either<ActionResult, ReleaseFileReference>(
-                        ValidationActionResult(ReplacementFileTypesMustBeData))
-                    : releaseFileReference);
+            return await _contentPersistenceHelper.CheckEntityExists<ReleaseFileReference>(
+                    id,
+                    q => q.Include(rfr => rfr.Release)
+                )
+                .OnSuccess<ActionResult, ReleaseFileReference, ReleaseFileReference>(
```

Comment:
Indeed. I wish C# would infer based on return types rather than solely on parameters. It kinda sucks because you get used to having that in Typescript!